src/test/java/org/jenkinsci/plugins/workflow/cps/DSLTest.java

```diff
@@ -164,6 +170,27 @@
         }
     }
 
+    @Test public void withCredentials() throws Exception {
+        final String credentialsId = "creds";
+        final String username = "bob";
+        final String password = "secr3t";
+        UsernamePasswordCredentialsImpl c = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, credentialsId, "sample", username, password);
+        CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), c);
+        WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+        String shellStep = Functions.isWindows()? "bat \"echo $PASSWORD\"\n" : "sh \"echo $PASSWORD\"\n";
+        p.setDefinition(new CpsFlowDefinition(""
+                + "node {\n"
+                + "withCredentials([usernamePassword(credentialsId: 'creds', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {\n"
+                + shellStep
+                + "  echo USERNAME\n"
+                + "  echo \"username is $USERNAME\"\n"
```

Comment:
I would probably just use `PASSWORD` in the tests, the fact that usernames are also considered secrets by `credentials-binding` is kind of problematic and may be changed at some point, see [JENKINS-44860](https://issues.jenkins-ci.org/browse/JENKINS-44860).

Reply:
addressed in b404458d32db852782e13c7bfc9b56458d9a80cc